spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java

```diff
@@ -1473,9 +1503,371 @@ static void queryWithQueryOptions(DatabaseClient dbClient) {
+          } else {
+            System.out.println(
+                String.format(
+                    "Database %s was restored from a backup and was optimized,"
+                    + "but the restore information no longer exists.",
+                    metadata.getName()));
+          }
+        } else {
+          System.out.println(
+              String.format(
+                  "Database %s was restored from a backup and was optimized,"
+                  + "but the database no longer exists.",
+                  metadata.getName()));
+        }
+      } catch (InvalidProtocolBufferException e) {
+        // The returned operation does not contain OptimizeRestoredDatabaseMetadata.
+        System.err.println(e.getMessage());
+      }
+    }
+    System.out.println(
+        String.format("Listed all optimize operations for database %s", databaseId.getDatabase()));
+  }
+  // [END spanner_list_database_operations]
+
+  // [START spanner_list_backups]
+  static void listBackups(InstanceAdminClient instanceAdminClient, DatabaseId database) {
+    Instance instance = instanceAdminClient.getInstance(database.getInstanceId().getInstance());
+    // List all backups.
+    System.out.println("All backups:");
+    for (Backup backup : instance.listBackups().iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups whose name contains 'users'.
+    System.out.println("All backups with backup name containing \"users\":");
+    for (Backup backup : instance.listBackups(Options.filter("name:users")).iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups for databases whose name contains the text 'bank'.
+    System.out.println("All backups for databases with a name containing \"bank\":");
+    for (Backup backup : instance.listBackups(Options.filter("database:bank")).iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups that expire before a certain time.
+    System.out.println("All backups that expire before 2019-10-18T02:56:53Z:");
+    for (Backup backup :
+        instance
+            .listBackups(Options.filter("expire_time < \"2019-10-18T02:56:53Z\""))
+            .iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups with size greater than a certain number of bytes.
+    System.out.println("All backups with size greater than 1000 bytes:");
+    for (Backup backup : instance.listBackups(Options.filter("size_bytes > 1000")).iterateAll()) {
+      System.out.println(backup);
+    }
+
+    // List all backups with a create time after a certain timestamp and that are also ready.
+    System.out.println("All databases created after 2019-10-18T02:56:53Z and that are ready:");
+    for (Backup backup :
+        instance
+            .listBackups(Options.filter("create_time >= \"2019-10-18T02:56:53Z\" AND state:READY"))
+            .iterateAll()) {
+      System.out.println(backup);
+    }
```

Comment:
I know this is how we defined it in the spec but we realized after that the samples tests won't be meaningful since these filter won't show anything. So we ended up doing the following:

1. Get the backup ID as input to listBackups and filter on name:backupId.
2. Database filter was changed to filter on the databaseId given as input.
3. Expire time filter was changed to expire_time < (time now + 30 days).
4. size_bytes > X where X depends on the size of the database you're creating in other parts of the test. PHP is using X = 500 and Node.js is using X = 100.
5. create_time >= (time now - 1 day)

Hope it's ok to make these changes.

Reply:
Done